From 05a818b27f0257281d43dcb8869c55f5a8e678b1 Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Thu, 30 May 2013 20:19:16 +0530
Subject: [PORT FROM MAIN][HDMI 5.1][REGRESSION] After 5.1 playback,touch tones are heard on IHF

BZ: 73360

The stream using direct output thread will not be stopped
or closed during stream pause. This will lead to maintanence of
the direct thread state after 5.1 playback. This is reset now
even before stream is closed on disconnect or close.

Change-Id: I3c2a4dfce957429fcbb25cf2b2766ddf5ff189fd
Orig-Change-Id: I246bf4509ebcf525b71ccfd495b548ef3caffd57
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 audio/AudioPolicyManagerBase.cpp |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index 4cce00a..e2ae86e 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -175,6 +175,10 @@ status_t AudioPolicyManagerBase::setDeviceConnectionState(audio_devices_t device
             return BAD_VALUE;
         }
 
+        /*set the direct output thread flag to false always. It will be
+          reset when the stream is started*/
+        mIsDirectOutputActive =  false;
+
         checkA2dpSuspend();
         checkOutputForAllStrategies();
         // outputs must be closed after checkOutputForAllStrategies() is executed
@@ -775,7 +779,7 @@ status_t AudioPolicyManagerBase::stopOutput(audio_io_handle_t output,
     AudioOutputDescriptor *outputDesc = mOutputs.valueAt(index);
 
     if((outputDesc->mFlags & AudioSystem::OUTPUT_FLAG_DIRECT) && (mAvailableOutputDevices & AudioSystem::DEVICE_OUT_AUX_DIGITAL)){
-       ALOGD("startoutput() Direct thread is stopped -inactive");
+       ALOGD("stopoutput() Direct thread is stopped -inactive");
        mIsDirectOutputActive = false;
     }
 
@@ -1860,6 +1864,9 @@ void AudioPolicyManagerBase::closeOutput(audio_io_handle_t output)
         }
     }
 
+    mIsDirectOutputActive =  false;
+    ALOGV("closeOtput: mIsDirectOutputActive = %d",mIsDirectOutputActive);
+
     AudioParameter param;
     param.add(String8("closing"), String8("true"));
     mpClientInterface->setParameters(output, param.toString());
-- 
1.7.4.1

